source/extensions/filters/common/lua/lua.cc

```diff
@@ -70,10 +70,14 @@ int ThreadLocalState::getGlobalRef(uint64_t slot) {
   return tls.global_slots_[slot];
 }
 
-uint64_t ThreadLocalState::registerGlobal(const std::string& global) {
-  tls_slot_->runOnAllThreads([global](OptRef<LuaThreadLocal> tls) {
+uint64_t ThreadLocalState::registerGlobal(const std::string& global,
+                                          const InitializerList& initializers) {
+  tls_slot_->runOnAllThreads([global, &initializers](OptRef<LuaThreadLocal> tls) {
```

Comment:
Sorry, my bad, seems like we need to copy the `initializers` vector.

```c++
tls_slot_->runOnAllThreads([global, initializers](OptRef<LuaThreadLocal> tls) {
```

Reply:
Thanks! Changed.